### DISCUSSION TITLE
Why does [up-layer] require a target when it's not a navigation?

### DISCUSSION THREAD

**Discussion by alexandergitter** [unpoly/unpoly#659] [Q&A] [answered]
Consider the following piece of code:

```html
<a href="/some/dialog" up-layer="new" up-navigate="false">
  Open modal dialog
</a>
```

This uses `up-navigate="false"` to turn off the navigation defaults.

When clicking this link, Unpoly throws an exception `up.CannotMatch: No target selector given` and I don't quite understand why. The returned content gets inserted into a new layer, what is the target used for in this case?

**Comment by triskweline** [comment, marked as answer]
> The returned content gets inserted into a new layer, what is the target used for in this case?

Unpoly allows the server to respond with a full HTML document. It needs to know which target selector to extract from the document and place it into the overlay.

With [navigation](https://unpoly.com/navigation#navigation-defaults) you get a default `{ fallback: ':main' }`, which extracts the [main element](https://unpoly.com/up-main) if no explicit target is given. The main element often works for overlays, as it usually excludes navigation bars and footers.
